ldap/servers/slapd/opshared.c

```diff
@@ -875,7 +875,11 @@ op_shared_search(Slapi_PBlock *pb, int send_result)
                  * sent a result */
                 rc = SLAPI_FAIL_GENERAL;
                 slapi_pblock_get(pb, SLAPI_RESULT_CODE, &err);
-                if (err == LDAP_NO_SUCH_OBJECT) {
+                if (err == LDAP_REFERRAL) {
+                    /* manage a referral search result code. */
+                    flag_referral++;
```

Comment:
I am not really fan about reusing this variable. Are you sure it has the same purpose as the one you need ?
May be you can create your own variable.